nepalingo-web/src/pages/TestYourself.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useEffect } from "react";
+import useDictionary from "@/hooks/useDictionary";
+import { generate } from "random-words";
+import Header from "@/components/header/Header";
+import Button from "@/components/Button";
+import { useStreak } from "@/hooks/StreakContext";
+import { useLanguage } from "@/hooks/Langauge";
+
+const TestYourself: React.FC = () => {
+  const { updateStreak } = useStreak();
+  const { selectedLanguage } = useLanguage();
+  const [word, setWord] = useState("today");
+  const [index, setIndex] = useState(0);
+
+  const [options, setOptions] = useState<string[]>([
+    "today",
+    "rice",
+    "hot",
+    "salt",
+  ]);
+  const [selectedOption, setSelectedOption] = useState<string | null>(null);
+  const [isCorrect, setIsCorrect] = useState<boolean | null>(null);
+  const { data, isLoading } = useDictionary({
+    language: selectedLanguage || "newari",
+    word,
+  });
+
+  useEffect(() => {
+    updateStreak(); // Trigger streak update on flashcard page load
+  }, [selectedLanguage]);
+
+  const getOptions = (word: string) => {
+    const randomWords = generate({ exactly: 3 }) as string[];
+    randomWords.push(word);
+    const shuffledOptions = randomWords.sort(() => Math.random() - 0.5);
+    setOptions(shuffledOptions);
+  };
+
+  function getNextIndex(wordArray: Array<string>) {
+    const newIndex = (index + 1) % wordArray.length;
+    return newIndex;
+  }
+
+  const handleNextQuestion = async () => {
+    let wordArray: Array<string> = [];
+    if (selectedLanguage === "Newari") {
```

Comment:
This word selection is the same as the Flashcard component, can be take this logic out and put it into a separate file with an exported function named something like `getNextWord(language)` so that we can share the logic. That way if we update the word list, we won't have to update it in both places.

Reply:
The flashcard page will also benefit from having the nextIndex logic and not being totally random.

Though I suggest simplifying the index as well and replacing it with a javascript generator https://www.programiz.com/javascript/generators#:~:text=To%20create%20a%20generator%2C%20you,generator%20const%20generator_obj%20%3D%20generator_function()%3B

So that you can just call the generatore.next() function to get the next word.